src/taipy/gui/gui.py

```diff
@@ -686,27 +686,48 @@ def _get_user_content_url(
         return f"/{Gui.__USER_CONTENT_URL}/{path or ''}?{urlencode(qargs)}"
 
     def __serve_user_content(self, path: str) -> t.Any:
-        if hasattr(self, "on_user_content") and callable(self.on_user_content):
-            self.__set_client_id_in_context()
+        self.__set_client_id_in_context()
+        qargs: t.Dict[str, str] = {}
+        qargs.update(request.args)
+        qargs.pop(Gui.__ARG_CLIENT_ID, None)
+        cb_function_name = qargs.get("custom_user_content_cb")
+        cb_function = None
+        if cb_function_name:
+            cb_function = self._get_user_function(cb_function_name)
+            if not callable(cb_function):
+                parts = cb_function_name.split(".", 2)
```

Comment:
From what I am seeing in the code we only make use of 2 items in the split. Max split should be 1 instead of 2 if you want a list of 2 elements according to [this](https://www.w3schools.com/python/ref_string_split.asp).